api/controllers/user.py

```diff
@@ -151,3 +151,78 @@ def get_user_profile(user_uuid: str, lang: Literal[CFG.UA, CFG.EN], db: Session)
         locations=locations,
         owned_rates_count=db_user.owned_rates_count,
     )
+
+
+def public_search_users(query: s.UserSearchIn, db: Session) -> s.PublicUsersSearchOut:
```

Comment:
What is the difference between public and non-public search?

Reply:
The difference is in the scheme, in the complexity of the filter and in the public filter we don't work with the current user - "me: m.User"
**def search_users(query: s.UserSearchIn, me: m.User, db: Session)**